### ISSUE TITLE
Unexpected backslash with OSC 9;9 prompt

### ISSUE THREAD

**Issue by yan12125** [PowerShell/PowerShell#19822]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

1. Configure OSC 9;9 for PowerShell and add custom prompt in `$PROFILE` following https://learn.microsoft.com/en-us/windows/terminal/tutorials/new-tab-same-directory#powershell-powershellexe-or-pwshexe:
```powershell
function prompt {
  $loc = $executionContext.SessionState.Path.CurrentLocation;

  $out = "PS $loc$('>' * ($nestedPromptLevel + 1)) ";
  if ($loc.Provider.Name -eq "FileSystem") {
    $out += "$([char]27)]9;9;`"$($loc.ProviderPath)`"$([char]27)\"
  }
  return $out
}
```
2. Type a single quote `'` or a dot `.`

### Expected behavior

```console
PS> '
```

### Actual behavior

```console
PS>\'
```

### Environment data

```powershell
PS /home/yen/tmp/PowerShell> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.4.0-preview.3
PSEdition                      Core
GitCommitId                    7.4.0-preview.3
OS                             Linux 6.3.8-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 14 Jun 2023 20:10:31 +0000
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

I can also reproduce the issue with PowerShell 7.3.4 on Windows 10 22H2.
```

### Visuals

![圖片](https://github.com/PowerShell/PowerShell/assets/1937689/e772f18e-0537-4022-ae75-6fe97f442b60)

This is captured with Konsole 23.04.2 and tmux 3.3a. I can reproduce the same issue with PowerShell 7.3.4 on Windows 10 22H2.

**Comment by yan12125** [comment]
The issue is gone after `Remove-module PSReadline`. Probably an issue with PSReadline?

Both PSReadline 2.2.6 and 2.3.1-beta1 have this issue.

**Comment by 237dmitry** [comment]
I had the exactly problem. In my case `OSC 9;9` output the last symbol of prompt string, `m` in prompt below. I do not know why and how but I fixed this by exporting [int] class (no matter what class).

```powershell
Set-PSReadlineOption -ContinuationPrompt $("`u{20}" * 5)
[console]::WriteLine("`e]12;#0087ff`a")
 
function Global:Prompt
{
    $null = [int]::new()          # No artefacts. I do not know why.
    $prompt = "`e[38;5;31m `u{ff05} `e]9;9;${PWD}`e\`e[0m"
    return $prompt
}
```

**Comment by yan12125** [comment]
You may want this instead of hacking around `$null`:
```powershell
$normalPrompt = "`e[38;5;31m `u{ff05} `e[0m"
$errorPrompt = "`e[91m `u{ff05} `e[0m"
Set-PSReadlineOption -PromptText $normalPrompt, $errorPrompt
```

From a comment in https://github.com/MicrosoftDocs/terminal/issues/597#issuecomment-1346749542,

> IMPORTANT: Make sure there's a printable charater _last_ in the prompt.
> Otherwise, PSReadline is gonna use the terminating `\` here and colorize
> that if it detects a syntax error

`-PromptText` explicitly specifies prompts to use for syntax error indication and avoids fragile auto detection.

I have no idea why hacking around `$null` works, though.

**Comment by yan12125** [comment]
Apparently PSReadLine simply cuts a substring since the last non-whitespace character in the prompt: https://github.com/PowerShell/PSReadLine/blob/v2.2.6/PSReadLine/ReadLine.cs#L819. It will be great if OSC commands are parsed and ignored, but that may be far from trivial due to the need of a parser of various escape sequences.

If I were a PowerShell developer, I might close this as WONTFIX :)

**Comment by 237dmitry** [comment]
> You may want this

No, I don't. I have shown this example in order to present inexplicable workaround for me.
